### PR TITLE
[FLINK-2391]Fix Storm-compatibility FlinkTopologyBuilder.createTopology bug

### PR DESCRIPTION
**Pull request by ffbin**
1.Error Scene:
Error happend in program like this:
builder.setSpout("source0", new Generator(pt), pt.getInt("sourceParallelism"));
builder.setBolt("sa", new RepartPassThroughBolt(pt), pt.getInt("sinkParallelism")).fieldsGrouping("source0", new Fields("id"));
builder.setBolt("sink", new Sink(pt), pt.getInt("sinkParallelism")).fieldsGrouping("sa", new Fields("id"));
final FlinkLocalCluster cluster = FlinkLocalCluster.getLocalCluster();
cluster.submitTopology("throughput", conf, builder.createTopology());
if the last bolt use fieldsGrouping, createTopology will throw NullPointerException.
2.Reason:
where get streaming group attribute index, it get downstream operator outputFields,this is error。Because the last bolt has no
outputFields, so the outputSchema of declarer in null and throw NullPointerException.
3.Modify:
Save operator declarer in one hashmap. Get producer declarer from hashmap when get upstream operator outputFields.

### CODE REVIEW

File: flink-contrib/flink-storm-compatibility/flink-storm-compatibility-core/src/main/java/org/apache/flink/stormcompatibility/api/FlinkTopologyBuilder.java [hunk 1]
```diff
@@ -153,7 +157,8 @@ public FlinkTopology createTopology() {
 							// global grouping is emulated in Storm via an empty fields grouping list
 							final List<String> fields = grouping.get_fields();
 							if (fields.size() > 0) {
-								inputDataStream = inputDataStream.groupBy(declarer.getGroupingFieldIndexes(grouping
+								FlinkOutputFieldsDeclarer procDeclarer = this.declarers.get(producerId);
```
Comment (mjsax) [inline flink-contrib/flink-storm-compatibility/flink-storm-compatibility-core/src/main/java/org/apache/flink/stormcompatibility/api/FlinkTopologyBuilder.java]: typo: prodDeclarer (not procDeclarer)

**Comment by mjsax** [comment]
LGTM. Let's see if Travis passes. Before merging, you need to update your commit message to be meaningful.

**Comment by mjsax** [comment]
Your commit message is still not ok. You update the GitHub description, but not the commit message. It is still "modify". You need to do a `git commit --amend -m "New commit message"`. The commit message should relate to the JIRA, ie, "[FLINK-2391] Storm-compatibility:method FlinkTopologyBuilder.createTopology() throws java.lang.NullPointerException"

**Comment by rmetzger** [comment]
By the way, you don't need to open another pull request for updating it.
Just force-push into the branch this PR is based on (`Feature2`).

**Comment by ffbin** [comment]
Thank you very much!

**Comment by mjsax** [comment]
+1 for merging

**Comment by StephanEwen** [comment]
Will merge this...

**Comment by mjsax** [comment]
Thanks for reporting the bug and providing a fix.

**Pull request closed without merging**